subprojects/core-api/src/main/java/org/gradle/api/artifacts/ModuleVersionSelector.java

```diff
@@ -67,4 +67,15 @@
      */
     @Incubating
     boolean matchesStrictly(ModuleVersionIdentifier identifier);
+
+    /**
+     * The module identifier of the component. Returns the same information
+     * as {@link #getGroup()} and {@link #getName()}.
+     *
+     * @return the module identifier
+     *
+     * @since 4.9
+     */
+    @Incubating
+    ModuleIdentifier getModule();
```

Comment:
I feel like naming this method `getModule()` instead of `getModuleIdentifier()` might be adding confusion as there are a number of places where a `getModule()` method actually means `getName()` such as in `ModuleComponentIdentifier`.
This comment also applies to the change in `DependencyMetadata`.

Reply:
The idea is more, IMO, to eventually get rid of the method that returns the "name" for `getModule`, because this one is confusing (it doesn't return the module, it returns the module name).